### PR TITLE
Pre-processing: Adding DivC and SubC kernels.

### PR DESCRIPTION
**Pull request by anna-khakimova**
1. DivC kernel.
2. SubC kernel.
3. Tests for new kernels.

### CODE REVIEW

File: inference-engine/tests_deprecated/fluid_preproc/common/fluid_tests.cpp [hunk 1]
```diff
@@ -678,6 +679,65 @@ TEST_P(ConvertDepthTestGAPI, AccuracyTest)
...
+    DivCComputation cc(to_test(in_mat1), to_test(out_mat_gapi), to_test(C));
+    cc.warmUp();
+
+    // OpenCV code /////////////////////////////////////////////////////////////
+    {
+        out_mat_ocv = in_mat1 / C;
+    }
+    // Comparison //////////////////////////////////////////////////////////////
+    {
+        EXPECT_LE(cv::norm(out_mat_ocv, out_mat_gapi, cv::NORM_INF), tolerance);
```
Comment (sivanov-work) [inline inference-engine/tests_deprecated/fluid_preproc/common/fluid_tests.cpp:706]: as far i remember `EXPECT_LE` doesn't generate some fatal exception and make tests passed for FAIL also.
maybe ASSERT_LE is expected here?

File: inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp [hunk 2]
```diff
@@ -12,8 +12,42 @@
...
+          m_v_out(std::move(v_out))
+    {}
+
+    Priv(cv::GComputation && c, cv::gapi::own::Mat&& v_in, cv::gapi::own::Mat&& v_out)
+        : m_c(std::move(c)),
+          m_v_in{std::move(v_in)},
+          m_v_out{std::move(v_out)}
+    {}
+
+    Priv(cv::GComputation && c, cv::gapi::own::Mat&& v_in, std::vector<cv::gapi::own::Mat>&& v_out)
```
Comment (sivanov-work) [inline inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp:30]: I could be wrong but `std::initializer_list<>` can reduce multiple constructor definistions

     Priv(..., std::initializer_list<own::Mat> v_in, std::initializer_list<own::Mat> v_out) :
      ...
       m_v_in(v_in),
       m_v_out(v_out)
     { }

**Review by sivanov-work — COMMENTED** [review]
LGTM

**Review by sivanov-work — APPROVED** [review]

File: inference-engine/tests_deprecated/fluid_preproc/common/fluid_tests.cpp [hunk 3]
```diff
@@ -678,6 +679,65 @@ TEST_P(ConvertDepthTestGAPI, AccuracyTest)
...
+    cc.warmUp();
+
+    // OpenCV code /////////////////////////////////////////////////////////////
+    {
+        out_mat_ocv = in_mat1 - C;
+    }
+    // Comparison //////////////////////////////////////////////////////////////
+    {
+        EXPECT_LE(cv::norm(out_mat_ocv, out_mat_gapi, cv::NORM_INF), tolerance)
+//                << "OCV: \n" << out_mat_ocv
```
Comment (rgarnov) [inline inference-engine/tests_deprecated/fluid_preproc/common/fluid_tests.cpp:734]: Please remove

File: inference-engine/tests_deprecated/fluid_preproc/common/fluid_tests.cpp [hunk 4]
```diff
@@ -1268,5 +1328,33 @@ TEST_P(PreprocTest, Performance)
             out_layout_str.c_str(), out_size.width, out_size.height,
             colorFormatToString(in_fmt).c_str(), colorFormatToString(out_fmt).c_str());
 #endif // PERF_TEST
+}
+
+TEST_P(MeanValueGAPI, AccuracyTest)
+{
+    const auto params = GetParam();
+//    int in_depth      = std::get<0>(params);
```
Comment (rgarnov) [inline inference-engine/tests_deprecated/fluid_preproc/common/fluid_tests.cpp:1336]: Please remove

File: inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp [hunk 5]
```diff
@@ -12,8 +12,42 @@
 struct FluidComputation::Priv
 {
     cv::GComputation m_c;
-    std::vector<cv::gapi::own::Mat> m_v_in;
+    cv::GRunArgs m_v_in;
     std::vector<cv::gapi::own::Mat> m_v_out;
+#if 1
```
Comment (rgarnov) [inline inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp:17]: Please remove

File: inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp [hunk 6]
```diff
@@ -27,19 +61,23 @@ namespace
 
 void FluidComputation::warmUp(test::Rect roi)
 {
-    if (roi.empty())
-        m_priv->m_c.apply(m_priv->m_v_in, m_priv->m_v_out, cv::compile_args(InferenceEngine::gapi::preprocKernels()));
-    else
-        m_priv->m_c.apply(m_priv->m_v_in, m_priv->m_v_out, cv::compile_args(InferenceEngine::gapi::preprocKernels(), cv::GFluidOutputRois{{to_own(roi)}}));
+    auto compile_args = roi.empty() ? cv::compile_args(InferenceEngine::gapi::preprocKernels())
+                                   :  cv::compile_args(InferenceEngine::gapi::preprocKernels(), cv::GFluidOutputRois{{to_own(roi)}});
```
Comment (rgarnov) [inline inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp:65]: Indentation

File: inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp [hunk 7]
```diff
@@ -220,8 +258,57 @@ ConvertDepthComputation::ConvertDepthComputation(test::Mat inMat, test::Mat outM
...
+                               })
+{}
+
+DivCComputation::DivCComputation(test::Mat inMat, test::Mat outMat, test::Scalar const& c)
+    : FluidComputation(new Priv{ []()-> cv::GComputation{
+                                    cv::GMat in;
+                                    cv::GScalar C;
+                                    cv::GMat out = in / C;
+                                    return cv::GComputation(cv::GIn(in, C), cv::GOut(out));
+                                }()
```
Comment (rgarnov) [inline inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp:272]: Please add one more whitespace here to align with lambda's declaration's start indentation

**Review by rgarnov — CHANGES_REQUESTED** [review]
Looks good, I just left some minor cosmetic comments

**Comment by anton-potapov** [comment]
@anna-khakimova how does it differ from #5240 ?

**Comment by anna-khakimova** [comment]
@anton-potapov This patch contains 2 fixes for 2 compilation issues on VS2019 compiler and CentOS76 CI build(old version of GCC compiler). Also it contains changes for succefully passing new unit tests.

File: inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp (hunk 5 above)
Comment (anna-khakimova) [reply inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp:17]: Done.

File: inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp (hunk 2 above)
Comment (anna-khakimova) [reply inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp:30]: Each constructors has different types of arguments. So in my opinion, std::initializer_list<> don't fit for this case.

File: inference-engine/tests_deprecated/fluid_preproc/common/fluid_tests.cpp (hunk 1 above)
Comment (anna-khakimova) [reply inference-engine/tests_deprecated/fluid_preproc/common/fluid_tests.cpp:706]: ASSERT_XXX(): If assertion fails then processing of test terminate.
EXPECT_XXX(): nonfatal failure, allowing processing to continue.
Here it is not a significant difference.

File: inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp (hunk 7 above)
Comment (anna-khakimova) [reply inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp:272]: Done.

File: inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp (hunk 6 above)
Comment (anna-khakimova) [reply inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp:65]: Done.

File: inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp (hunk 5 above)
Comment (anna-khakimova) [reply inference-engine/tests_deprecated/fluid_preproc/fluid_test_computations/fluid_test_computations.cpp:17]: Done.

File: inference-engine/tests_deprecated/fluid_preproc/common/fluid_tests.cpp (hunk 4 above)
Comment (anna-khakimova) [reply inference-engine/tests_deprecated/fluid_preproc/common/fluid_tests.cpp:1336]: Done.

File: inference-engine/tests_deprecated/fluid_preproc/common/fluid_tests.cpp (hunk 3 above)
Comment (anna-khakimova) [reply inference-engine/tests_deprecated/fluid_preproc/common/fluid_tests.cpp:734]: Done.

**Comment by anna-khakimova** [comment]
@rgarnov please take a look.

**Comment by anna-khakimova** [comment]
@ilyachur , @ilya-lavrenov please take a look.

**Review by rgarnov — APPROVED** [review]
Looks good, thanks!

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**